cmd/blockchaincmd/add_validator.go

```diff
@@ -548,14 +547,24 @@ func CallAddValidator(
 	if err != nil {
 		return err
 	}
-	if err = signatureaggregator.UpdateSignatureAggregatorPeers(app, network, extraAggregatorPeers, aggregatorLogger); err != nil {
-		return err
-	}
-	signatureAggregatorEndpoint, err := signatureaggregator.GetSignatureAggregatorEndpoint(app, network)
-	if err != nil {
-		return err
+	if signatureAggregatorEndpoint == "" {
+		signatureAggregatorEndpoint, err = signatureaggregator.GetSignatureAggregatorEndpoint(app, network)
+		if err != nil {
+			// if local machine does not have a running signature aggregator instance for the network, we will create it first
+			err = signatureaggregator.CreateSignatureAggregatorInstance(app, network, aggregatorLogger, "latest")
+			if err != nil {
+				return err
+			}
+			signatureAggregatorEndpoint, err = signatureaggregator.GetSignatureAggregatorEndpoint(app, network)
+			if err != nil {
+				return err
+			}
+		}
 	}
+	aggregatorCtx, aggregatorCancel := sdkutils.GetTimedContext(constants.SignatureAggregatorTimeout)
+	defer aggregatorCancel()
 	signedMessage, validationID, rawTx, err := validatormanager.InitValidatorRegistration(
+		aggregatorCtx,
```

Comment:
this is not signature aggregator context. it is context used to do event searchs onto the blockchain.
maybe we can add a constant for evmEventLookupTimeout (2 minutes?) and create a specific context.
if not here, we need a ticket can you add it